addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -36,13 +124,199 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
+            id: 1,
+            display_name: "first record",
+            header: "",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="header" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+        await click(target, '.o_form_button_save');
+        assert.containsOnce(target, ".o_notification");
+        const notif = target.querySelector(".o_notification");
+        assert.strictEqual(
+            notif.querySelector(".o_notification_content").textContent,
+            "Header"
+        );
+        assert.hasClass(notif, "border-danger");
+
+    });
+
+    QUnit.test('colorpicker', async function (assert) {
+        assert.expect(10);
+
+        serverData.models['note.note'].records = [{
+            id: 1,
+            display_name: "first record",
+            header: "<p>  &nbsp;&nbsp;  <br>   </p>",
+            body: "<p>toto toto toto</p><p>tata</p>",
+        }];
+
+        // Add the ajax service (legacy), because wysiwyg RPCs use it.
+        patchWithCleanup(legacyEnv, {
+            services: {
+                ...legacyEnv.services,
+                ajax: {
+                    rpc: async (route, args) => {
+                        if (args.model === "ir.ui.view" && args.method === 'render_public_asset') {
+                            if (args.args[0] === "web_editor.colorpicker") {
+                                return COLOR_PICKER_TEMPLATE;
+                            }
+                            if (args.args[0] === "web_editor.snippets") {
+                                return SNIPPETS_TEMPLATE;
+                            }
+                        }
+                    },
+                },
+            }
+        });
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="body" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+
+        const field = target.querySelector('.o_field_html[name="body"]');
+        const editable = field.querySelector("[contenteditable='true']");
+        await click(editable);
+
+        assert.strictEqual(
+            window.getComputedStyle(document.querySelector("#toolbar")).getPropertyValue("visibility"),
+            "hidden",
+            "should hide the toolbar");
+
+        // select the text
+        const pText = editable.querySelector("p").firstChild
+        Wysiwyg.setRange(pText, 1, pText, 10);
+        // text is selected
+
+        var range = Wysiwyg.getRange();
+
+        assert.strictEqual(range.sc, pText,
+            "should select the text");
+
+        await wait(1);
```

Comment:
What is this ? Why is it there ?

Reply:
It's to allow for async stuff to happen, before continuing the test.

Reply:
I suggest you try `await nextTick()` instead.